pkg/canary/reader/reader.go

```diff
@@ -280,7 +292,11 @@ func (r *Reader) Query(start time.Time, end time.Time) ([]time.Time, error) {
 	}
 	req.Header.Set("User-Agent", userAgent)
 
-	resp, err := http.DefaultClient.Do(req)
+	httpClient, err := r.httpClient()
+	if err != nil {
+		return nil, err
```

Comment:
I would highly recommend wrapping the error here. (and also on the `QueryCountOverTime`) with some context.

Rationale is I see we use it on both `Query` and `QueryCountOverTime` and it can confuse the consumer of the error, where exactly it is coming from.